src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2366,6 +2367,9 @@ protected virtual void MarkReflectionLikeDependencies (MethodBody body)
 			if (HasManuallyTrackedDependency (body))
 				return;
 
+			var scanner = new ReflectionMethodBodyScanner (this);
```

Comment:
Would you mind giving us some way to turn this new code off without us having to customize our fork?

One way would be refactoring `MarkReflectionLikeDependencies` a bit so that we can override the new behavior while leaving the existing reflection detection running.

```
		protected virtual void MarkReflectionLikeDependencies (MethodBody body)
		{
			if (HasManuallyTrackedDependency (body))
				return;

                       MarkReflectionWithScanner(body);
                       MarkReflectionWithPatternDetector(body);
		}
        
        protected virtual void MarkReflectionWithScanner(MethodBody body)
        {
             var scanner = new ReflectionMethodBodyScanner (this);
             scanner.Scan (body);
        }
        
        protected virtual void MarkReflectionWithPatternDetector(MethodBody body)
        {
        	       var instructions = body.Instructions;
			ReflectionPatternDetector detector = new ReflectionPatternDetector (this, body.Method);

			//
			// Starting at 1 because all patterns require at least 1 instruction backward lookup
			//
			for (var i = 1; i < instructions.Count; i++) {
				var instruction = instructions [i];

				if (instruction.OpCode != OpCodes.Call && instruction.OpCode != OpCodes.Callvirt)
					continue;

				if (ProcessReflectionDependency (body, instruction))
					continue;

				if (!(instruction.Operand is MethodReference methodCalled))
					continue;

				var methodCalledDefinition = methodCalled.Resolve ();
				if (methodCalledDefinition == null)
					continue;

				ReflectionPatternContext reflectionContext = new ReflectionPatternContext (_context, body.Method, methodCalledDefinition, i);
				try {
					detector.Process (ref reflectionContext);
				}
				finally {
					reflectionContext.Dispose ();
				}
			}
        }
```

Alternatively, some sort of bool option on `LinkContext` to disable usage of `ReflectionMethodBodyScanner ` would work as well.

There is a lot of new code here and no tests.  I'm sure it was tested else where, but I don't want this new code to get in the way of us staying in sync with upstream monolinker

Reply:
I should have made it more clear - this is going into a branch so that we can shape it into something acceptable for master. Being able to turn this off, making sure it doesn't run unless absolutely necessary, etc. is going to be addressed before this would make it into master.

We're going to add tests too - there are existing targeted tests we can port and we're going to write new tests too.

Reply:
O great that sounds good.

We have more complicated reflection detection tests in our UnityLinker test suite that are supported by stack analysis that we have.  I can put these tests on branch for you to grab if you’d like.

Reply:
We would be very grateful for any tests in this area. Thanks a lot!

Reply:
@vitek-karas here are all of our reflection tests 

https://github.com/Unity-Technologies/linker/commit/1c37396edcb3edeaccf3dc96530c023a8202c892

You can cherry-pick this to a branch from master and everything should run.  Of course many tests will fail.